cwltool/job.py

```diff
@@ -903,19 +906,25 @@ def _job_popen(
     timelimit: Optional[int] = None,
     name: Optional[str] = None,
     monitor_function=None,  # type: Optional[Callable[[subprocess.Popen[str]], None]]
+    default_stdin: Optional[Union[IO[Any], int]] = None,
+    default_stdout: Optional[IO[Any]] = None,
+    default_stderr: Optional[IO[Any]] = None,
 ) -> int:
 
     if job_script_contents is None and not FORCE_SHELLED_POPEN:
 
-        stdin = subprocess.PIPE  # type: Union[IO[Any], int]
+        stdin = default_stdin if default_stdin is not None else subprocess.PIPE # type: Union[IO[Any], int]
```

Comment:
I don't think this makes sense, users are not supposed to ever interact directly with running programs

Reply:
I did it to not have [issue](https://github.com/common-workflow-language/cwltool/issues/221) with the process. To be able to use the runtimeContext like this:

runtime_context.default_stdin = subprocess.DEVNULL
runtime_context.default_stdout = subprocess.DEVNULL
runtime_context.default_stderr = subprocess.DEVNULL

https://github.com/giannisdoukas/CWLJNIKernel/blob/master/cwlkernel/CoreExecutor.py#L47

Reply:
Maybe I'm misunderstanding the problem, but what the original code is doing is that stdin for the subprocess is always a pipe, it never reads from sys.stdin.  You don't want it to read from sys.stdin either, and the existing code already does that.  I'm suggesting that `default_stdin` is unnecessary -- `default_stdout` and `default_stderr` are fine.

Reply:
Yes, it was my fault because I did not include any test to illustrate how I want to use that feature. To be honest I wasn't sure if I had to do it for the stdin but I wanted for sure for the stdout/stderr.

Reply:
So I think you want to revert the stdin changes, and add a test case for capturing stdout/stderr.